server/testrun/testrun.go

```diff
@@ -347,42 +353,61 @@ finishedloop:
 
 }
 
-func cleanTestData(dirtyData map[string]string) map[string]map[string]map[string]float32 {
+// cleanTestData cleans up the testing data that comes back from the agents.
+// The test data that comes back raw from the agents is "dirty", meaning it is designed to be as flexible
+// as possible.
+func cleanTestData(dirtyData map[string]string) (map[string]map[string]map[string]interface{}, error) {
```

Comment:
Eh, `MarshalJSON()` is defined on `*json.RawMessage`, so the changes to `executetestrun.go` would need to be:
``` go
gatheredData := map[string]*json.RawMessage{}
```
and
``` go
b := json.RawMessage(cmdOutput.Bytes())
gatheredData[thisTarget] = &b
```